apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DataAdapterTests.cs

```diff
@@ -0,0 +1,135 @@
+                Assert.Equal(2, testDataSet.Tables[0].Columns.Count);
+                Assert.Equal(_testFixture.TestDatabaseRowCount, testDataSet.Tables[0].Rows.Count);
+            }
+        }
+
+        [Fact]
+        public async Task AdapterDeleteInsert()
+        {
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                var adapter = new SpannerDataAdapter(connection, _testFixture.DataAdapterTestTable, "Key");
+
+                //Load
+                var testDataSet = new DataSet();
+                adapter.Fill(testDataSet);
+                var k0 = testDataSet.Tables[0].Rows[0]["Key"];
+                var stringValue0 = testDataSet.Tables[0].Rows[0]["StringValue"];
+                Assert.IsType<string>(k0);
+                Assert.IsType<string>(stringValue0);
+
+                //Delete, reload
+                testDataSet.Tables[0].Rows[0].Delete();
+                adapter.Update(testDataSet);
+                testDataSet.Clear();
+                adapter.Fill(testDataSet);
+                Assert.Equal(_testFixture.TestDatabaseRowCount - 1, testDataSet.Tables[0].Rows.Count);
+
+                //insert, reload
+                var newRow = testDataSet.Tables[0].NewRow();
+                newRow["Key"] = k0;
+                newRow["StringValue"] = stringValue0;
+                testDataSet.Tables[0].Rows.Add(newRow);
+                adapter.Update(testDataSet);
+                testDataSet.Clear();
+                adapter.Fill(testDataSet);
+                Assert.Equal(_testFixture.TestDatabaseRowCount, testDataSet.Tables[0].Rows.Count);
+            }
+        }
+
+        [Fact]
+        public async Task AdapterUpdate()
+        {
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                var adapter = new SpannerDataAdapter(connection, _testFixture.DataAdapterTestTable, "Key");
+
+                //Load
+                var testDataSet = new DataSet();
+                adapter.Fill(testDataSet);
+
+                //update, reload
+                var newValue = Guid.NewGuid().ToString();
+                testDataSet.Tables[0].Rows[1]["StringValue"] = newValue;
+                adapter.Update(testDataSet);
+                testDataSet.Clear();
+                adapter.Fill(testDataSet);
+                Assert.Equal(newValue, testDataSet.Tables[0].Rows[1]["StringValue"]);
+            }
+        }
+
+        [Fact]
+        public async Task AdapterOverrideSelect()
+        {
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                var adapter = new SpannerDataAdapter(connection, _testFixture.DataAdapterTestTable, "Key")
+                {
+                    SelectCommand =
+                        connection.CreateSelectCommand(
+                            $"SELECT * FROM {_testFixture.DataAdapterTestTable} WHERE Key='k1'")
```

Comment:
actually thats why I chose k1 -- k1 is actually the second row and isnt touched by another test.
There are still some interdependencies between the tests -- eg if the delete/insert fails halfway thru, it can leave the table with 14 rows where it should be 15 and cause the read test to fail.
Its not a big deal though.